### PR TITLE
Setup GH Action to support multi-arch for cilium/packaging

### PR DESCRIPTION
**Pull request by Jianlin-lv**
Enable "GitHub-hosted runners + buildx" to build multi-arch image;
Create script support build images and push manifest.

secrets.DOCKERHUB_USERNAME: repository  user
secrets.DOCKERHUB_PASSWORD: repository  password
Be set from Settings->Secrets->Add a new secret.

Signed-off-by: Jianlin Lv <Jianlin.Lv@arm.com>

### CODE REVIEW

File: .github/workflows/build_container.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      run: |
+        ./build_container/docker_manifest.sh cilium-llvm `date +'%Y-%m-%d'`
+        ./build_container/docker_manifest.sh cilium-iproute2 `date +'%Y-%m-%d'`
+        ./build_container/docker_manifest.sh cilium-bpftool `date +'%Y-%m-%d'`
+
+
+  # build arm64 images
+  jobs-arm64:
+    continue-on-error: true
+    runs-on: [self-hosted, Linux, ARM64]
```
Comment (errordeveloper) [inline .github/workflows/build_container.yml:32]: Nice, didn't realise these were supported. What is availability like?

File: .github/workflows/build_container.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - name: create manifest
+      run: |
+        ./build_container/docker_manifest.sh cilium-llvm `date +'%Y-%m-%d'`
+        ./build_container/docker_manifest.sh cilium-iproute2 `date +'%Y-%m-%d'`
+        ./build_container/docker_manifest.sh cilium-bpftool `date +'%Y-%m-%d'`
+
+
+  # build arm64 images
+  jobs-arm64:
+    continue-on-error: true
```
Comment (errordeveloper) [inline .github/workflows/build_container.yml:31]: Why is this needed?

File: .github/workflows/build_container.yml (hunk 1 above)
Comment (errordeveloper) [reply .github/workflows/build_container.yml:32]: Or will this need to be a custom runner?

Comment (Jianlin-lv) [reply .github/workflows/build_container.yml:32]: yes,  we need to prepare arm64 environment,  Self-hosted runners can be physical, virtual, in a container, or in a cloud,
The machine can communicate with GitHub Actions;
Add runner  from  Settings->actions->add runner   page
Please refer to https://github.blog/2019-11-05-self-hosted-runners-for-github-actions-is-now-in-beta/

File: .github/workflows/build_container.yml (hunk 2 above)
Comment (Jianlin-lv) [reply .github/workflows/build_container.yml:31]: you mean "continue-on-error: true". 

I'm not sure if the community has an Arm64 running environment; 
so add this too prevents a workflow run from failing when arm64 job fails.

File: .github/workflows/build_container.yml (hunk 1 above)
Comment (errordeveloper) [reply .github/workflows/build_container.yml:32]: I see, and have you tried using binfmt and qemu?

Here is what I did earlier: https://github.com/errordeveloper/buildx-test/blob/3e1ca27f81688f43bd4e6dfe420494ea4e5a81c1/.github/workflows/images.yaml#L12-L16

File: .github/workflows/build_container.yml (hunk 2 above)
Comment (errordeveloper) [reply .github/workflows/build_container.yml:31]: I see, I guess it makes sense to have this to begin with. Would you mind adding a comment to clarify the intent?

File: .github/workflows/build_container.yml (hunk 1 above)
Comment (Jianlin-lv) [reply .github/workflows/build_container.yml:32]: I have tried to compile cilium image using Qemu on a bare-metal machine, it is worked,
Have not tried the "GH action + binfmt" method;
Does the community prefer to use "GitHub-hosted + binfmt" to build the multi-arch image

Comment (errordeveloper) [reply .github/workflows/build_container.yml:32]: Yeah, there is currently some infra that is sponsored and managed by Isovalent, and the aim is to move as much as possible to GitHub, so it's maintained as part of the project.
Having to host a custom GitHub Action runner would introduce a new component that someone would have to maintain, it's best to avoid that. Of course, test clusters are going to be hosted in AWS, GCP and other providers, but image builds shouldn't depend on anything outside GitHub.

Comment (Jianlin-lv) [reply .github/workflows/build_container.yml:32]: Make sense, I will update this PR again recently.

**Comment by Jianlin-lv** [comment]
One issue is Dockerfile.llvm execution time  more than 6 hours, The job is terminated,
I'm configuring a longer time for workflow and be testing on my private repo.
